Sources/BraintreeCore/Configuration/ConfigurationLoader.swift

```diff
@@ -33,29 +35,28 @@ class ConfigurationLoader {
     /// 2. If no cached configuration is found, it fetches the configuration from the server and caches the successful response.
     /// 3. If fetching the configuration fails, it returns an error.
     ///
-    /// - Parameters:
-    ///   - completion: A completion handler that is called with the fetched or cached `BTConfiguration` object or an `Error`.
-    ///
-    /// - Completion:
-    ///   - `BTConfiguration?`: The configuration object if it is successfully fetched or retrieved from the cache.
-    ///   - `Error?`: An error object if fetching the configuration fails or if the instance is deallocated.
+    /// - Returns: A `BTConfiguration` if it is successfully fetched or retrieved from the cache.
+    /// - Throws: An `Error` describing the failure; if fetching the configuration fails or if the instance is deallocated.
     @_documentation(visibility: private)
     @ConfigurationActor
     func getConfig() async throws -> BTConfiguration {
+        if let cachedConfig = try? configurationCache.getFromCache(authorization: http.authorization.bearer) {
+            return cachedConfig
+        }
+
+        /// if we are writing to the cache at this time, we can return the existing task
         if let existingTask {
             return try await existingTask.value
         }
 
-        if let cachedConfig = try? configurationCache.getFromCache(authorization: http.authorization.bearer) {
-            return cachedConfig
-        }
-     
-        let task = Task { [weak self] in
+        existingTask = Task { [weak self] in
```

Comment:
I think like Sammy suggested we do not need a new property here that we then set to the existing task. I went through several flows to ensure we weren't reintroducing any bugs but think we should be covered here. It will also reduce confusion from the previous PR where we were setting `existingTask = task` but then using `task.value`

Reply:
I think it's same logic, I just thought it was easier to read that way that a task is being created there, maybe for debugging also, but I am not attached to it.